Package.swift

```diff
@@ -24,7 +28,6 @@ let package = Package(
             dependencies: [
                 .product(name: "Collections", package: "swift-collections"),
                 .product(name: "Logging", package: "swift-log"),
-                .product(name: "NIO", package: "swift-nio"),
```

Comment:
Note, there is a way to perform this elision for non-wasm builds only, if desired. This seemed like the best solution, though, since it appears that only NIOCore is being used currently in this repository.

Reply:
Per @gwynne 's comment below about fluent kit downstream problems, I decided to narrow the scope of this change to wasm-only platforms.